### ISSUE TITLE
Core - CefResourceRequestHandlerAdapter.GetCookieAccessFilter object reference not set to an instance of an object.

### ISSUE THREAD

**Issue by GrabzIt** [cefsharp/CefSharp#3967]
- **What version of the product are you using?**
97.1.11

- **What architecture x86 or x64?**
Any
	
- **What version of .Net?**
Any
    
- **On what operating system?**
Any

- **Are you using `WinForms`, `WPF` or `OffScreen`?**
Any

- **Please provide any additional information below.**

```
Object reference not set to an instance of an object.
   at CefSharp.Internals.CefResourceRequestHandlerAdapter.GetCookieAccessFilter(CefResourceRequestHandlerAdapter* , scoped_refptr<CefCookieAccessFilter>* , scoped_refptr<CefBrowser>* browser, scoped_refptr<CefFrame>* frame, scoped_refptr<CefRequest>* request) in C:\projects\cefsharp\CefSharp.Core.Runtime\Internals\CefResourceRequestHandlerAdapter.h:line 62
```

I think this is occurring here in my code:

```c#
        protected override CefReturnValue OnBeforeResourceLoad(IWebBrowser browserControl, IBrowser browser, IFrame frame, IRequest request, IRequestCallback callback)
        {
            if (string.IsNullOrEmpty(parameters.InputFilePath) && request.ResourceType == ResourceType.MainFrame)
            {
                if (capture.CookieJar != null)
                {
                    foreach (UniversalCookie cookie in capture.CookieJar.GetCookies())
                    {
                        browserControl.GetCookieManager().SetCookieAsync(request.Url, cookie.GetCefCookie());
                    }
                }
            }
```

This isn't a consistent error btw sometimes it will work fine.

**Comment by amaitland** [comment]
Thanks for the stack trace. I'll add some null checks.

Any further insight on how to reproduce the problem would be greatly appreciated.

**Comment by GrabzIt** [comment]
Thanks I am unsure why it occurs it happens for some  websites and not others. I am pretty sure that the code snippet is where it is occuring. I suspect some sort of race condition might be involved.

**Comment by amaitland** [comment]
For resource requests that are handled by [CefRequestContextHandlerAdapter::GetResourceRequestHandler](https://github.com/cefsharp/CefSharp/blob/cefsharp/97/CefSharp.Core.Runtime/Internals/CefRequestContextHandlerAdapter.cpp#L33) then the `IWebBrowser` reference passed into [CefResourceRequestHandlerAdapter will be nullptr](https://github.com/cefsharp/CefSharp/blob/cefsharp/97/CefSharp.Core.Runtime/Internals/CefRequestContextHandlerAdapter.cpp#L61) This wasn't accounted for in commit https://github.com/cefsharp/CefSharp/commit/4ade0b8284d4f95f763c471a867139bff2bac2a9 

I've added additional checks before calling TryGetBrowserCoreById

**Comment by amaitland** [comment]
This should be fixed in version `97.1.12` which has been pushed to `Nuget.org` and should appear shortly.

Please let me know if you have any further problems.

**Comment by GrabzIt** [comment]
Thanks that's awesome!
